### DISCUSSION TITLE
Apply -composite to an entire folder

### DISCUSSION THREAD

**Discussion by johnb2233** [ImageMagick/ImageMagick#1952] [General]
I am a newbie to IM I have 7.0.8 Q16 (64-bit) installed on WIndows 10 Pro 64bit

As part of a workflow I use the following in a 10 batch file
```cmd
cd /D D:\fold1
 
for /f %%a in ('dir *.jpg /b') do magick D:\fold2\black.jpg  %%a -geometry +144+108 -composite -quality 100  %%a
```

I am using a loop to overlay each image (in D:\fold1) on a black (1920x1080) background, this works but is very slow as invokes IM for each file

Is it possible to write this command as single command rather than invoking IM, in this case 200, times?

I have tried several options but with no success.

**Comment by snibgo** [comment]
You are also re-reading black.jpg every time. (JPEG compression is lossy, so that file may not be exactly the colours you want.) Use a .mpc or .miff file instead to increase performance. If that file should be entirely black, then you could use "-size 1920x1080 xc:Black" instead.

I don't think you can use "-composite" in "magick mogrify".

Another possibility is to create a long command, something like:

magick ^
D:\fold2\black.jpg -write mpr:BCK +delete ^
( mpr:BCK in1.jpg -geometry +144+108 -composite -quality 100 -write in1.jpg +delete ) ^
( mpr:BCK in2.jpg -geometry +144+108 -composite -quality 100 -write in2.jpg +delete ) ^
( mpr:BCK in3.jpg -geometry +144+108 -composite -quality 100 -write in3.jpg +delete ) ^
:
:
exit

You can create this long command with a loop. Build the command in an environment variable, or in a BAT script file, or in an ImageMagick script file (without the line-continuation ^ characters).

**Reply by johnb2233** [reply]
Many thanks for your response and apologies for my slow response back. I have been unable to get back to this problem, due to other commitments.  I will try this method as soon as I am able.  Again thanks for your response.

**Comment by fmw42** [comment]
You can do a composite in magick mogrify use -draw to compose the black.jpg under the other images. See https://imagemagick.org/Usage/basics/#mogrify and https://imagemagick.org/Usage/basics/#mogrify_compose

The other way is to read all the input images (assuming you have the RAM) to hold them and then use -layers composite similar to compositing with an animation image. See https://imagemagick.org/Usage/anim_mods/#composite and https://imagemagick.org/Usage/compose/#dstin

`magick image1 ... imageN null: black.jpg -compose dst_in -layers composite result_%d.jpg
`

The latter will not keep the input names for the output. Magick mogrify will and is the usual way to do that.

**Reply by johnb2233** [reply]
Many thanks for your response and apologies for my slow response back. I have been unable to get back to this problem, due to other commitments.  I will try this method as soon as I am able.  Again thanks for your response.
